source/custom/nimxlcterminaladaptor_restricted_converters.h

```diff
@@ -31,20 +29,39 @@ struct NIErrStatusOutputConverter {
     return &status;
   }
 
+  std::string UrlEncode(char* input) const
+  {
+    std::string encoded(input);
+    // Replace the following characters to their equivalent '%xx' hex code
+    const char* charsToEncode = "!*'();:@&=+$,/?%#[] <>~";
```

Comment:
Is there a library for this?

I'm not sure if we can reuse https://github.com/grpc/grpc/blob/master/src/core/lib/slice/percent_encoding.h but you should look at which characters it encodes.

Reply:
Updated to follow the method used by percent_encoding.cc to determine which characters to encode.